tests/pyrog/api/test_credential.py

```diff
@@ -140,13 +106,11 @@ def test_filter_credentials_by_source(
             None,
             None,
             "POSTGRES",
-            200,
+            400,
```

Comment:
this test is not parametrized for the "success" case anymore ?

Reply:
I think this test excepts a 400 bc the patched credentials are not valid anymore (random int).

Reply:
* fun fact: there is 0.1% that this test will failed
* whether we should keep this behaviour is debatable since it will not mean that the credentials will stay valid (they could be changed). But this question applies to every model representing the source DB, so we could tackle this one later

Reply:
mmh no the issue is we're using patch here and the validate method in the credential serializer takes only the request data and has no access to the instance previously recorded. So either I don't allow a patch or I will need to apply the validation in both the create and update method separately.